### PR TITLE
Correct documentation suggesting invalid maxCorners value

### LINKED ISSUE #2: Exception if `ref_img` is not larger than `mon_img` in box axes
If `ref_img` is smaller than the `mon_img` in either X or Y, line 255 in `matcher.py` fails to read any data, as x_off+x_size and y_off+y_size exceed the raster bounds:
```
ref_box = ref_img.read(1, x_off, y_off, x_size, y_size)
```

`ref_box` is returned as None, and an exception is raised on line 234, because this is an invalid operation on `None`:
```
[...] & np.isfinite(ref_box) & [...]
```

The documentation never states that `ref_img` must be larger.

If this is a bug, a quick fix would be to modify the `x_size` and `y_size` calculations to always refer to only the valid area across both rasters:
```
        max_x = min(mon_img.x_size, ref_img.x_size)
        max_y = min(mon_img.y_size, ref_img.y_size)
        x_size = (
            self._conf.tile_size
            if x_off + self._conf.tile_size < max_x
            else max_x - x_off
        )
        y_size = (
            self._conf.tile_size
            if y_off + self._conf.tile_size < max_y
            else max_y - y_off
        )
```
but an ideal solution might be to read the maximum area from each raster, if the tiles can be of different sizes.

Separately, I note that Python exceptions are not enabled in GDAL (by calling `gdal.UseExceptions()`), and the code does not explicitly check that GDAL calls succeeded. This is what leads to the `ref_img.read` call failing silently, and the unhelpful exception being raised later. GDAL is already emitting log warnings about this, and best practice would be to enable exceptions.

### LINKED ISSUE #3: `max_corners < 0` is not a valid input, despite documentation
The README states that:

> maxCorners : Maximum number of corners to extract. If there are more corners than are found, the strongest of them is returned.
> **maxCorners <= 0** implies that no limit on the maximum is set and all detected corners are returned.

This suggests that an input such as `"maxCorners": -1` in the configuration is valid. However, doing so fails, because `opencv` does not permit negative values for this parameter:
```
Traceback (most recent call last):
  File "/home/danielevans/repositories/karios/karios/karios.py", line 319, in <module>
    sys.exit(main(sys.argv[1:]))
             ^^^^^^^^^^^^^^^^^^
  File "/home/danielevans/repositories/karios/karios/karios.py", line 313, in main
    match_and_plot.process(args.mon, args.ref, args.mask, args.resume)
  File "/home/danielevans/repositories/karios/karios/karios.py", line 280, in process
    points = self._get_points(resume, monitored_image, reference_image, mask)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/danielevans/repositories/karios/karios/karios.py", line 148, in _get_points
    points = self._do_klt(monitored_image, reference_image, mask, csv_file)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/danielevans/repositories/karios/karios/karios.py", line 127, in _do_klt
    points = self._handle_klt_results(dataframe_gen, csv_file)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/danielevans/repositories/karios/karios/karios.py", line 64, in _handle_klt_results
    for dataframe in results:
  File "/home/danielevans/repositories/karios/karios/klt_matcher/matcher.py", line 200, in match
    points = self._match_tile(x_off, y_off, mon_img, ref_img, mask)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/danielevans/repositories/karios/karios/klt_matcher/matcher.py", line 261, in _match_tile
    results = klt_tracker(
              ^^^^^^^^^^^^
  File "/home/danielevans/repositories/karios/karios/klt_matcher/matcher.py", line 100, in klt_tracker
    p0 = cv2.goodFeaturesToTrack(ref_data, mask=mask, **feature_params)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
cv2.error: OpenCV(4.8.1) /home/conda/feedstock_root/build_artifacts/libopencv_1698890291961/work/modules/imgproc/src/featureselect.cpp:389: error: (-215:Assertion failed) qualityLevel > 0 && minDistance >= 0 && maxCorners >= 0 in function 'goodFeaturesToTrack'
```

Using a value of 0 works as expected.

### PR DESCRIPTION
**Pull request by DFEvans**
Fixes #3 - the README suggests that maxCorners < 0 is a valid value, but OpenCV only permits values >= 0.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -5,7 +5,7 @@ repos:
    - id: fix-encoding-pragma
    - id: check-docstring-first
 - repo: https://github.com/johann-petrak/licenseheaders.git
-  rev: 'master'
```
Comment (DFEvans) [inline .pre-commit-config.yaml:8]: A quick drive-by fix - `pre-commit` was upset that this was not set to a fixed revision, so unexpected changes could be brought in.

**Review by telespazio-tim — APPROVED** [review]
Thanks Daniel
